### PR TITLE
Allow environment variables and constants as default values

### LINKED ISSUE #1525: VIP: Allow environment variables and constants as "default value" in public functions
There are a number of Uniswap functions that have a recipient field. It would be great to default this field to msg.sender like so:

```python
@public
@payable
def ethToTokenSwap(minTokens: uint256, recipient: address = msg.sender):
```

Currently I have to do something along these lines:
```python
@public
@payable
def ethToTokenSwap(minTokens: uint256, recipient: address = 0x000...000):
   if(recipient == ZERO_ADDRESS):
      recipient = msg.sender
```

which isn't the worst but it would be great to have msg.sender, msg.value, and constants available here.

### PR DESCRIPTION
**Pull request by iamdefinitelyahuman**
### What I did
Allow builtin constants and environment variables to be used as defaults for keyword arguments - closes #1525 

This does NOT allow custom constants.

### How I did it
* `vyper/parser/expr.py` - added `BUILTIN_CONSTANTS` and `ENVIRONMENT_VARIABLES`
* `vyper/signatures/function_signature.py` - add checks to `validate_default_values`

### How to verify it
Run the updated tests.

### Description for the changelog
* Allow constants and environment variables to be used as defaults for keyword arguments

### Cute Animal Picture
image

### CODE REVIEW

File: tests/parser/functions/test_default_parameters.py [hunk 1]
```diff
@@ -158,6 +160,39 @@ def callMeMaybe() -> (bytes[100], uint256, bytes[20]):
...
+
+
+def test_environment_vars_as_default(get_contract):
+    code = """
+xx: uint256(wei)
+
+@public
+@payable
+def foo(a: uint256(wei) = msg.value) -> bool:
+    self.xx = a
```
Comment (fubuloubu) [inline tests/parser/functions/test_default_parameters.py]: Do `+=` here instead and call this twice in the test to show proper operation

File: tests/parser/functions/test_default_parameters.py [hunk 2]
```diff
@@ -190,7 +190,7 @@ def bar() -> uint256(wei):
     c.foo(transact={'value': 31337})
     assert c.bar() == 31337
     c.foo(666, transact={'value': 9001})
-    assert c.bar() == 666
+    assert c.bar() == 31337 + 666
```
Comment (fubuloubu) [inline tests/parser/functions/test_default_parameters.py]: Maybe also add a line that says "balance of `c` contract is `31337 + 9001`" to show the difference in using default args.

File: vyper/parser/expr.py [hunk 3]
```diff
@@ -53,6 +53,23 @@
     string_to_bytes,
 )
 
+# var name: (lllnode, type)
+BUILTIN_CONSTANTS = {
+    'EMPTY_BYTES32': ([0], 'bytes32'),
```
Comment (charles-cooper) [inline vyper/parser/expr.py]: style-wise I think the list constructor can be factored out here.

**Comment by charles-cooper** [comment]
LGTM

**Pull request merged**